Added invoke docs (#1769)
software-mansion/protostar: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/invoke.md

# `invoke`

```cairo
fn invoke(contract_address: felt252, function_name: felt252, calldata: Array::<felt252>) -> Result::<(), felt252> nopanic;
```

> Invokes a contract. `function_name` parameter should be provided as a shortstring.

Reviewer: ```suggestion
Invokes a contract. `function_name` parameter should be provided as a short string.
```

> Invokes a contract. `function_name` parameter should be provided as a shortstring.

Reviewer 2: As per @karol-bisztyga suggestion:

```suggestion
Invokes a contract. `function_name` parameter should be provided as a shortstring. It can mutate the state of the contract and does not return a value.  
```

Reviewer 2: And also we could consider mentioning that to query the contract for values you must use `call`?

Author: done

```cairo title="Example"
use result::ResultTrait;
use array::ArrayTrait;

fn my_function() {

>     let address = 123;

Reviewer 2: ```suggestion
    let contract_address = 123;
```

> Invokes a contract. `function_name` parameter should be provided as a short string. `invoke` can mutate the state of the contract and does not return a value, to query the contract for values use [call](./call.md). 

Reviewer: ```suggestion
Invokes a contract's function. `function_name` parameter should be provided as a short string. `invoke` can mutate the state of the contract and does not return a value, to query the contract for values use [call](./call.md). 
```

    let mut calldata = ArrayTrait::new();
    calldata.append(10);
    calldata.append(11);
    calldata.append(12);

>     invoke(address, 'function_name', calldata).unwrap();

Author: ```suggestion
    invoke(contract_address, 'function_name', calldata).unwrap();
```
